### PR TITLE
test: create system tests

### PR DESCRIPTION
**Pull request by JustinBeckwith**
This way `npm test` will always work locally, and system-tests are only run in CircleCI.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -14,6 +14,10 @@ install_and_test: &install_and_test
         name: Test
         command: npm test
         when: always
+    - run:
+        name: System Tests
+        command: npm run system-tests
+        when: always
```
Comment (stephenplusplus) [inline .circleci/config.yml]: Maybe this shouldn’t be always, so that if decrypt fails, this step will be skipped.

**Comment by JustinBeckwith** [comment]
Hmm, this is weird.  If you're just storing the key in an env var inside of CircleCI, I think I should be able to get it with a fork *scratches head*

@ofrobots @jinwoo @jmdobry @jonwayne y'all have ideas here?

**Comment by jinwoo** [comment]
I'm not familiar with this repo, but it seems that previously tests were mocking requests but your system tests are not and getting auth errors. Am I mistaken?

**Comment by stephenplusplus** [comment]
As in, you think it should expose it to forks? It’s an option that I have disabled so that it’s not abused.

**Comment by stephenplusplus** [comment]
@jinwoo we have 3 tests that make real requests, otherwise, we block all of them with stubbed methods.

**Comment by JustinBeckwith** [comment]
@stephenplusplus if we could enable it for forks that would be super convenient :)

**Comment by stephenplusplus** [comment]
I think CI is working as best it can for now. Do you know if there’s a “set -e” or something command we can use in the “command” section of the decrypt step? That way, the line could fail, but wouldn’t exit with an error code?

**Comment by JustinBeckwith** [comment]
Well the thing is, we want to make sure we're not breaking these tests before merging right? A few things to try:
- We could enable the env var for forks/PRs
- We could git rid of the e2e-edness, and mock the HTTP requests

**Comment by stephenplusplus** [comment]
This is the same pattern followed by the GC libs. If we allow forks to have access to the secret, then that means we give everyone my service account JSON file.

**Comment by stephenplusplus** [comment]
System tests still don’t run before merges for GC, we actually have to run locally or wait for the master merge to fail. Solutions we’ve talked about before would be having a whitelist of users whose PRs will have system tests run for, or some way to integrate with the GH review API, so that after we approve, it runs them. I’m not sure if either are possible or how difficult it would be to implement. But, just ideas.

**Comment by JustinBeckwith** [comment]
::nods:: sounds like this is a generalizable problem we need to solve beyond this PR.  For now - how about I start submitting PRs to this repo from other branches in the repository.  That way (I think?) CI should run.  If we can land this PR, then `npm test` will still work without these tests :)

How about we land this, then I'll re-submit #34 on another branch in this repo?

**Comment by stephenplusplus** [comment]
Here’s a related convo: https://github.com/GoogleCloudPlatform/google-cloud-common/issues/222#issuecomment-289064962

**Comment by ofrobots** [comment]
Exposing private credentials to PRs would be a security risk. Please don't do it. For most contributions unit tests on the PR and system tests on landing is the policy that people generally use.

For developers of a module, using local branches for PR is a good work-around.

**Pull request merged**